ArduPlane/mode_autoland.cpp

```diff
@@ -152,14 +164,39 @@ bool ModeAutoLand::_enter()
 void ModeAutoLand::update()
 {
     plane.calc_nav_roll();
+
+    // Constrain the roll limit if climb before turn options are set when below that altitude,
+    //that way if something goes wrong the plane will
+    // eventually turn back and go to landing waypoint instead of going perfectly straight. This also leaves
+    // some leeway for fighting wind.
+    check_altitude();
+    uint16_t roll_limit_cd;
+    if (!done_climb) {
+        roll_limit_cd = MIN(plane.roll_limit_cd, plane.g.level_roll_limit*100);
+        plane.nav_roll_cd = constrain_int16(plane.nav_roll_cd, -roll_limit_cd, roll_limit_cd);
+        // set CLIMB before turn alt as target if needed
+        if (climb_before_turn_alt > final_wp_alt) {
+            plane.target_altitude.amsl_cm = plane.home.alt + climb_before_turn_alt * 100.0f;
```

Comment:
I would have expected this in the enter not update.

Reply:
works there too I think...will try